### ISSUE TITLE
[Feature Request] Add option for including source props in jsx elements

### ISSUE THREAD

**Issue by MasterOfBob777** [evanw/esbuild#1119]
**What would you like esbuild to do?**
I'd like to add an option where it could generate source props on jsx elements similar to [babel's jsx transform source plugin](https://babeljs.io/docs/en/babel-plugin-transform-react-jsx-source) (also while you're at it maybe this one too, [self prop](https://babeljs.io/docs/en/babel-plugin-transform-react-jsx-self), it's of the same vein).

**Why is unfit for a plugin?**
Currently the only way to get source props with esbuild is to run babel first with the plugin specified earlier, and well not only is babel slow, it has to do a lot of things that esbuild already does (generate ast, transform it, generate code, etc.) and it uses a good amount of patterns that aren't esbuild incompatible per se but they don't really mesh well. And this isn't a niche use case either, many jsx based frameworks have devtools that support this format.

**tl;dr**
this 
```jsx
<sometag />
```
to this
```jsx
<sometag 
    __source={ { fileName: 'this/file.js', lineNumber: 10, columnNumber: 1 } } 
    __self={this} 
/>
```

**Comment by evanw** [comment]
Sorry but I consider this to be out of scope. I do not plan for esbuild to be a general AST-transformation tool like this. I agree that someone needs to build a Babel replacement that's actually fast, but that's a huge project by itself and it's beyond the scope of what I want esbuild to be.

You may want to take a look at [swc](https://swc.rs/). I haven't used it myself but I believe it's a reimplementation of a Babel-inspired AST in Rust with a focus on speed. So I imagine a transform like this would be very much in scope for that project.

**Comment by evanw** [comment]
Closing for the reason described above: this is out of scope.
